src/orquestra/sdk/_base/_driver/_client.py

```diff
@@ -119,33 +132,33 @@ class DriverClient:
     Client for interacting with the Workflow Driver API via HTTP.
     """
 
-    def __init__(self, base_uri: str, session: requests.Session):
-        self._base_uri = base_uri
+    def __init__(self, session: requests.Session, uri_provider: ExternalUriProvider):
+        self._uri_provider = uri_provider
         self._session = session
 
     @classmethod
-    def from_token(cls, base_uri: str, token: str):
+    def from_token(cls, token: str, uri_provider: ExternalUriProvider):
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** ‏same here